docs/architecture/dispatch.md

```diff
@@ -0,0 +1,64 @@
+# Dispatch
+
+The **dispatch** function controls how to execute [actions](actions.md) and apply the state transitions and effect running that they request to be done.
+
+Custom dispatchers are loosely comparable to middleware used in other frameworks with the difference being that they are just intended to augment the regular dispatching process for specialized purposes you may have for your project, particularly in relation to dev tooling.
+
+---
+
+## Dispatch Initializer
+
+The dispatch initializer accepts the default dispatch as its sole argument and must give back a dispatch in return. Hyperapp's default dispatch initializer is equivalent to:
+
+```js
+const boring = (dispatch) => dispatch
+```
+
+Of course, in your own initializer you'll likely want to return a variant of the regular dispatch.
+
+---
+
+## Custom Dispatching
+
+A dispatch function accepts an [action](actions.md) and its [payload](actions.md#payloads) if it has one. The action will then be carried out and its resulting state transition will be applied and then any effects it requested to be run will be run.
+
+```js
+const dispatch = (action, payload) => {
+  // Magic happens!
+}
+```
+
+Hyperapp's default dispatch function is a little too involved to showcase here but suffice it to say you'll most likely want to leverage it when crafting your own dispatching.
+
+---
+
+## Example
+
+Let's say we're working on a project where we want to always log the current state. Creating a custom dispatch for this purpose makes sense because it lets us accomplish this without having to change any of our existing actions to request our logging effect to run.
+
+```js
+import { log } from "./fx"
+
+const middleman = (dispatch) => (action, payload) => {
```

Comment:
I suggest renaming `middleman` to `middleware` here.

Reply:
Yeah, "middleware" is better here.  "middleman" was a pun but I suppose it's a bit obtuse especially when the rest of the code example feels regular.